### PR TITLE
Async and high-level refactoring for `bittensor/utils` module.

### PR DESCRIPTION
**Pull request by basfroman**
Async and high-level refactoring for `bittensor/utils` module.

### CODE REVIEW

File: bittensor/commands/delegates.py [hunk 1]
```diff
@@ -691,7 +691,7 @@ class ListDelegatesCommand:
 
     The table columns include:
 
-    - INDEX: The delegate's index in the sorted list.
+    - INDEX: The delegate's index in the sorted_ list.
```
Comment (camfairchild) [inline bittensor/commands/delegates.py:694]: why the underscore?

File: bittensor/utils/__init__.py [hunk 2]
```diff
@@ -231,17 +226,17 @@ def ss58_address_to_bytes(ss58_address: str) -> bytes:
...
-def U64_NORMALIZED_FLOAT(x: int) -> float:
+def u64_normalized_float(x: int) -> float:
     return float(x) / float(U64_MAX)
 
 
 def u8_key_to_ss58(u8_key: List[int]) -> str:
-    r"""
-    Converts a u8-encoded account key to an ss58 address.
+    """
+    Converts an u8-encoded account key to a ss58 address.
```
Comment (camfairchild) [inline bittensor/utils/__init__.py]:
```suggestion
    Converts a u8-encoded account key to an ss58 address.
```

File: bittensor/commands/delegates.py (hunk 1 above)
Comment (basfroman) [inline bittensor/commands/delegates.py]: answered on the channel

File: bittensor/utils/_register_cuda.py [hunk 3]
```diff
@@ -57,20 +57,19 @@ def _hex_bytes_to_u8_list(hex_bytes: bytes):
         ]
         return hex_chunks
 
-    def _create_seal_hash(block_and_hotkey_hash_hex: bytes, nonce: int) -> bytes:
+    def _create_seal_hash(block_and_hotkey_hash_hex_: bytes, nonce: int) -> bytes:
```
Comment (camfairchild) [inline bittensor/utils/_register_cuda.py:60]: Not sure the trailing underscore is needed here

File: bittensor/utils/registration.py [hunk 4]
```diff
@@ -536,7 +562,7 @@ async def _solve_for_difficulty_fast(
     log_verbose: bool = False,
 ) -> Optional[POWSolution]:
     """
-    Solves the POW for registration using multiprocessing.
+    Solves the POW for reg istration using multiprocessing.
```
Comment (camfairchild) [inline bittensor/utils/registration.py]:
```suggestion
    Solves the POW for registration using multiprocessing.
```

File: bittensor/utils/weight_utils.py [hunk 5]
```diff
@@ -61,7 +61,7 @@ def normalize_max_weight(
         if estimation.max() <= limit:
             return weights / weights.sum()
 
-        # Find the cumulative sum and sorted tensor
+        # Find the cumulative sum and sorted_ tensor
```
Comment (camfairchild) [inline bittensor/utils/weight_utils.py]:
```suggestion
        # Find the cumulative sum and sorted tensor
```

File: bittensor/utils/_register_cuda.py (hunk 3 above)
Comment (basfroman) [reply bittensor/utils/_register_cuda.py:60]: block_and_hotkey_hash_hex used outside the scope of _create_seal_hash function

File: bittensor/utils/formatting.py [hunk 6]
```diff
@@ -1,15 +1,55 @@
...
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+# the Software.
+#
+# THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+"""Modules provides the functionality to formatting of data."""
```
Comment (camfairchild) [inline bittensor/utils/formatting.py]:
```suggestion
"""Modules provides the functionality for formatting data."""
```

File: bittensor/commands/delegates.py [hunk 7]
```diff
@@ -51,7 +51,7 @@ def show_delegates_lite(
     """
     This method is a lite version of the :func:`show_delegates`. This method displays a formatted table of Bittensor network delegates with detailed statistics to the console.
 
-    The table is sorted by total stake in descending order and provides
+    The table is sorted_ by total stake in descending order and provides
```
Comment (camfairchild) [inline bittensor/commands/delegates.py]:
```suggestion
    The table is sorted by total stake in descending order and provides
```

File: bittensor/commands/delegates.py [hunk 8]
```diff
@@ -160,7 +160,7 @@ def show_delegates(
     """
     Displays a formatted table of Bittensor network delegates with detailed statistics to the console.
 
-    The table is sorted by total stake in descending order and provides
+    The table is sorted_ by total stake in descending order and provides
```
Comment (camfairchild) [inline bittensor/commands/delegates.py]:
```suggestion
    The table is sorted by total stake in descending order and provides
```

File: bittensor/commands/delegates.py [hunk 9]
```diff
@@ -616,7 +616,7 @@ class ListDelegatesLiteCommand:
 
     The table columns include:
 
-    - INDEX: The delegate's index in the sorted list.
+    - INDEX: The delegate's index in the sorted_ list.
```
Comment (camfairchild) [inline bittensor/commands/delegates.py]:
```suggestion
    - INDEX: The delegate's index in the sorted list.
```

File: bittensor/commands/delegates.py (hunk 1 above)
Comment (camfairchild) [inline bittensor/commands/delegates.py]:
```suggestion
    - INDEX: The delegate's index in the sorted list.
```

**Review by camfairchild — APPROVED** [review]

**Review by thewhaleking — APPROVED** [review]
I'm good with this, but why did we remove `utils/subtensor.py`?

**Comment by basfroman** [comment]
> I'm good with this, but why did we remove `utils/subtensor.py`?

I created `bittensor/utils/subtensor.py` to store functionality for handling errors from metadata. but it was decided to use a different way to retrieving those errors. after all this is unused code.

**Pull request merged**